hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/H2LegacyDialect.java

```diff
@@ -0,0 +1,707 @@
+import static org.hibernate.type.SqlTypes.ARRAY;
+import static org.hibernate.type.SqlTypes.BINARY;
+import static org.hibernate.type.SqlTypes.CHAR;
+import static org.hibernate.type.SqlTypes.DECIMAL;
+import static org.hibernate.type.SqlTypes.DOUBLE;
+import static org.hibernate.type.SqlTypes.FLOAT;
+import static org.hibernate.type.SqlTypes.GEOMETRY;
+import static org.hibernate.type.SqlTypes.INTERVAL_SECOND;
+import static org.hibernate.type.SqlTypes.LONG32NVARCHAR;
+import static org.hibernate.type.SqlTypes.LONG32VARBINARY;
+import static org.hibernate.type.SqlTypes.LONG32VARCHAR;
+import static org.hibernate.type.SqlTypes.NCHAR;
+import static org.hibernate.type.SqlTypes.NUMERIC;
+import static org.hibernate.type.SqlTypes.NVARCHAR;
+import static org.hibernate.type.SqlTypes.TIMESTAMP_UTC;
+import static org.hibernate.type.SqlTypes.UUID;
+import static org.hibernate.type.SqlTypes.VARBINARY;
+import static org.hibernate.type.SqlTypes.VARCHAR;
+
+/**
+ * A legacy {@linkplain Dialect SQL dialect} for H2.
+ *
+ * @author Thomas Mueller
+ */
+public class H2LegacyDialect extends Dialect {
+	private static final CoreMessageLogger LOG = CoreLogging.messageLogger( H2LegacyDialect.class );
+
+	private final LimitHandler limitHandler;
+
+	private final boolean ansiSequence;
+	private final boolean cascadeConstraints;
+	private final boolean useLocalTime;
+
+	private final SequenceInformationExtractor sequenceInformationExtractor;
+	private final String querySequenceString;
+
+	public H2LegacyDialect(DialectResolutionInfo info) {
+		this( parseVersion( info ) );
+		registerKeywords( info );
+	}
+
+	public H2LegacyDialect() {
+		this( SimpleDatabaseVersion.ZERO_VERSION );
+	}
+
+	public H2LegacyDialect(DatabaseVersion version) {
+		super(version);
+
+		// https://github.com/h2database/h2database/commit/b2cdf84e0b84eb8a482fa7dccdccc1ab95241440
+		limitHandler = version.isSameOrAfter( 1, 4, 195 )
+				? OffsetFetchLimitHandler.INSTANCE
+				: LimitOffsetLimitHandler.INSTANCE;
+
+		if ( version.isBefore( 1, 2, 139 ) ) {
+			LOG.unsupportedMultiTableBulkHqlJpaql( version.getMajor(), version.getMinor(), version.getMicro() );
+		}
+
+//		supportsTuplesInSubqueries = version.isSameOrAfter( 1, 4, 198 );
+
+		// Prior to 1.4.200 there was no support for 'current value for sequence_name'
+		// After 2.0.202 there is no support for 'sequence_name.nextval' and 'sequence_name.currval'
+		ansiSequence = version.isSameOrAfter( 1, 4, 200 );
+
+		// Prior to 1.4.200 the 'cascade' in 'drop table' was implicit
+		cascadeConstraints = version.isSameOrAfter( 1, 4, 200 );
+		// 1.4.200 introduced changes in current_time and current_timestamp
+		useLocalTime = version.isSameOrAfter( 1, 4, 200 );
+
+		if ( version.isSameOrAfter( 1, 4, 32 ) ) {
+			this.sequenceInformationExtractor = version.isSameOrAfter( 1, 4, 201 )
+					? SequenceInformationExtractorLegacyImpl.INSTANCE
+					: SequenceInformationExtractorH2DatabaseImpl.INSTANCE;
```

Comment:
Actually there is no such version as 1.4.201. The next release after 1.4.200 is 2.0.202.

Maybe I'm missing something, but why `SequenceInformationExtractorLegacyImpl.INSTANCE` is used for newer versions and `SequenceInformationExtractorH2DatabaseImpl.INSTANCE` is used for older versions in this condition?

Reply:
If you have ideas to improve the dialect, please create an issue for that and provide a PR